packages/astro/test/test-utils.js

```diff
@@ -52,7 +55,10 @@ export async function loadFixture(inlineConfig) {
 
   return {
     build: (opts = {}) => build(config, { mode: 'development', logging: 'error', ...opts }),
-    startDevServer: () => dev(config, { logging: 'error' }),
+    startDevServer: () => {
+      config.devOptions.port = currentPort++;
```

Comment:
I’m not sure if this will keep track of the port correctly because the dev server may shift (also ports may not be claimed in the order that this fn is called). The `dev()` function below should return a public `.port` property that has the correct port. It should take an open port now; it didn’t before.

Reply:
Good call, could be why OSX is failing, I'll check it out.